ts/src/pro/bitget.ts

```diff
@@ -993,43 +993,33 @@ export default class bitget extends bitgetRest {
          */
         await this.loadMarkets ();
         let market = undefined;
-        let marketId = undefined;
-        let isTrigger = undefined;
-        [ isTrigger, params ] = this.isTriggerOrder (params);
-        let messageHash = (isTrigger) ? 'triggerOrder' : 'order';
-        let subscriptionHash = 'order:trades';
+        let type = undefined;
+        let subType = undefined;
+        let settle = undefined;
+        let instId = undefined;
+        let instType = undefined;
         if (symbol !== undefined) {
             market = this.market (symbol);
             symbol = market['symbol'];
-            marketId = market['id'];
-            messageHash = messageHash + ':' + symbol;
+            instId = market['id'];
         }
-        const productType = this.safeString (params, 'productType');
-        let type = undefined;
-        [ type, params ] = this.handleMarketTypeAndParams ('watchOrders', market, params);
-        let subType = undefined;
-        [ subType, params ] = this.handleSubTypeAndParams ('watchOrders', market, params, 'linear');
-        if ((type === 'spot' || type === 'margin') && (symbol === undefined)) {
+        [ params, type, subType, settle, instType ] = this.handleProductTypesAndParams (market, 'watchOrders', params);
```

Comment:
This method is very very confusing, and the breaks the pattern of the handleXAndParams that usually returns the `[option, params]` and not a bunch of options